src/substrait/textplan/TypeConversion.cpp

```diff
@@ -0,0 +1,151 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include "substrait/textplan/TypeConversion.h"
+
+#include "substrait/type/Type.h"
+
+namespace io::substrait::textplan {
+
+namespace {
+
+std::string kNullSignature = "?";
+
+} // namespace
+
+// Converts a Type proto directly into its text signature form.
+std::string typeToText(const ::substrait::proto::Type& type) {
+  switch (type.kind_case()) {
+    case ::substrait::proto::Type::kBool:
+      if (type.bool_().nullability() ==
+          ::substrait::proto::Type::NULLABILITY_NULLABLE) {
+        return TypeTraits<TypeKind::kBool>::kSignature + kNullSignature;
```

Comment:
why use  kSignature as text string  for bool type here and kTypeString for other types?